mongo/src/test/java/org/axonframework/extensions/mongo/eventsourcing/eventstore/MongoEventStorageEngineTest_DBObjectSerialization.java

```diff
@@ -17,84 +17,63 @@
-import org.axonframework.extensions.mongo.MongoTemplate;
 import org.axonframework.extensions.mongo.serialization.DBObjectXStreamSerializer;
-import org.axonframework.extensions.mongo.utils.MongoLauncher;
 import org.axonframework.serialization.Serializer;
 import org.axonframework.serialization.upcasting.event.EventUpcaster;
 import org.junit.jupiter.api.*;
-import org.junit.jupiter.api.extension.*;
-import org.springframework.beans.factory.annotation.Autowired;
-import org.springframework.context.ApplicationContext;
-import org.springframework.context.annotation.Bean;
-import org.springframework.context.annotation.Configuration;
-import org.springframework.test.annotation.DirtiesContext;
-import org.springframework.test.context.ContextConfiguration;
-import org.springframework.test.context.junit.jupiter.SpringExtension;
-import org.springframework.transaction.PlatformTransactionManager;
-
-import java.io.IOException;
+import org.testcontainers.containers.MongoDBContainer;
+import org.testcontainers.junit.jupiter.Container;
+import org.testcontainers.junit.jupiter.Testcontainers;
 
-import static org.junit.jupiter.api.Assumptions.*;
-import static org.mockito.Mockito.*;
+import java.util.Collections;
 
 /**
  * Test class validating the {@link MongoEventStorageEngine} with the {@link DBObjectXStreamSerializer}.
  *
  * @author Rene de Waele
  */
-@DirtiesContext
-@ExtendWith(SpringExtension.class)
-@ContextConfiguration(classes = MongoEventStorageEngineTest_DBObjectSerialization.TestContext.class)
+@Testcontainers
 class MongoEventStorageEngineTest_DBObjectSerialization extends AbstractMongoEventStorageEngineTest {
 
-    private static MongodExecutable mongoExe;
-    private static MongodProcess mongod;
+    @Container
+    private static final MongoDBContainer MONGO_DB_CONTAINER = new MongoDBContainer("mongo");
+    private static final Serializer DB_OBJECT_XSTREAM_SERIALIZER = DBObjectXStreamSerializer.builder().build();
 
-    @Autowired
-    private ApplicationContext context;
     private DefaultMongoTemplate mongoTemplate;
-
     @SuppressWarnings("FieldCanBeLocal")
     private MongoEventStorageEngine testSubject;
 
-    @BeforeAll
-    static void start() throws IOException {
-        mongoExe = MongoLauncher.prepareExecutable();
-        mongod = mongoExe.start();
-    }
-
-    @AfterAll
-    static void shutdown() {
-        if (mongod != null) {
-            mongod.stop();
-        }
-        if (mongoExe != null) {
-            mongoExe.stop();
-        }
-    }
-
     @BeforeEach
     void setUp() {
-        MongoClient mongoClient = null;
-        try {
-            mongoClient = context.getBean(MongoClient.class);
-        } catch (Exception e) {
-            assumeTrue(true, "No Mongo instance found. Ignoring test.");
-        }
-        mongoTemplate = DefaultMongoTemplate.builder().mongoDatabase(mongoClient).build();
+        MongoSettingsFactory mongoSettingsFactory = new MongoSettingsFactory();
```

Comment:
Is it possible to move this to a different class and reuse this code?

Reply:
Think so. Some form of utility class I guess. Can have a look what that would look like.

Reply:
Made the suggested adjustment. Please let me know what you think!